File: app/models/coupon.py
from app.extensions import db

class Coupon(db.Model):
    __tablename__ = 'coupon'

    coupon_id = db.Column(db.BigInteger, primary_key=True, autoincrement=True)
    coupon_name = db.Column(db.String(255), nullable=False)
    coupon_discount_rate = db.Column(db.Integer, nullable=False)
    coupon_valid_days = db.Column(db.Integer, nullable=False)

File: src/main/java/com/ticketflow/entity/Coupon.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "coupon")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Coupon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "coupon_id")
    private Long couponId;

    @Column(name = "coupon_name", length = 255, nullable = false)
    private String couponName;

    @Column(name = "coupon_discount_rate", nullable = false)
    private Integer couponDiscountRate;

    @Column(name = "coupon_valid_days", nullable = false)
    private Integer couponValidDays;
}
